clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesCredentialsInitializer.groovy

```diff
@@ -0,0 +1,101 @@
+import org.apache.log4j.Logger
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.config.ConfigurableBeanFactory
+import org.springframework.context.ApplicationContext
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Scope
+import org.springframework.stereotype.Component
+
+@Component
+@Configuration
+class KubernetesCredentialsInitializer implements CredentialsInitializerSynchronizable {
+  private static final Logger log = Logger.getLogger(this.class.simpleName)
+
+  @Autowired
+  String kubernetesApplicationName
+
+  @Autowired
+  AccountCredentialsRepository accountCredentialsRepository
+
+  @Autowired
+  ApplicationContext appContext;
+
+  /*
+  @Autowired
+  List<ProviderSynchronizerTypeWrapper> providerSynchronizerTypeWrappers
+  */
+
+  @Bean
+  List<? extends KubernetesNamedAccountCredentials> kubernetesNamedAccountCredentials(
+    KubernetesConfigurationProperties kubernetesConfigurationProperties) {
+    synchronizeKubernetesAccounts(kubernetesConfigurationProperties, null)
+  }
+
+  @Override
+  String getCredentialsSynchronizationBeanName() {
+    return "synchronizeKubernetesAccounts"
+  }
+
+  @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+  @Bean
+  List<?> synchronizeKubernetesAccounts(KubernetesConfigurationProperties kubernetesConfigurationProperties, CatsModule catsModule) {
+    def (ArrayList<KubernetesConfigurationProperties.ManagedAccount> accountsToAdd, List<String> namesOfDeletedAccounts) =
+      ProviderUtils.calculateAccountDeltas(accountCredentialsRepository,
+                                           KubernetesNamedAccountCredentials,
+                                           kubernetesConfigurationProperties.accounts)
+
+    accountsToAdd.each { KubernetesConfigurationProperties.ManagedAccount managedAccount ->
+      try {
+        def kubernetesAccount = new KubernetesNamedAccountCredentials(managedAccount.name,
+                                                        managedAccount.environment ?: managedAccount.name,
+                                                        managedAccount.accountType ?: managedAccount.name,
+                                                        managedAccount.master,
+                                                        managedAccount.username,
+                                                        managedAccount.password,
+                                                        managedAccount.namespace,
+                                                        kubernetesApplicationName)
+
+        accountCredentialsRepository.save(managedAccount.name, kubernetesAccount)
+      } catch (e) {
+        log.info "Could not load account ${managedAccount.name} for Kubernetes.", e
+      }
+    }
+
+    ProviderUtils.unscheduleAndDeregisterAgents(namesOfDeletedAccounts, catsModule)
+
+    /*
```

Comment:
Please add a comment as to why this is commented-out, but not remove entirely. (I'm ok leaving it, just drop a comment.)